topic: does education and healthcare spending affect quality of life?

### dataset links

life expectancy dataset: https://data.worldbank.org/indicator/SP.DYN.LE00.IN

education: https://data.worldbank.org/indicator/SE.XPD.TOTL.GD.ZS

Government health spending as a share of GDP: https://ourworldindata.org/grapher/public-health-expenditure-share-gdp?country=SWE~FRA~DEU~JPN~GBR~ESP~AUS~NZL~CAN~USA~AUT~FIN~SVN~NOR~NLD~BEL~CHE~DNK~ISL~CZE~SVK~PRT~ARG~ITA~POL~CYP~COL~IND~IDN~CHN~MEX~TUR~THA~PER~ZAF~UKR~BRA~LVA~ROU~CRI~HUN~BGR~GRC~KOR~LUX~LTU~EST~HRV~MLT~CHL~IRL~ISR

### code

#### datasets

In [1]:
import pandas as pd

In [2]:
eduDf = pd.read_csv("API_SE.XPD.TOTL.GD.ZS_DS2_en_csv_v2_11528.csv")
lifeDf = pd.read_csv("API_SP.DYN.LE00.IN_DS2_en_csv_v2_11567.csv")
healthDf = pd.read_csv("public-health-expenditure-share-gdp.csv")

In [3]:
eduDf.head()

Country Name Country Code  \
0                        Aruba          ABW   
1  Africa Eastern and Southern          AFE   
2                  Afghanistan          AFG   
3   Africa Western and Central          AFW   
4                       Angola          AGO   

                                      Indicator Name     Indicator Code  1960  \
0  Government expenditure on education, total (% ...  SE.XPD.TOTL.GD.ZS   NaN   
1  Government expenditure on education, total (% ...  SE.XPD.TOTL.GD.ZS   NaN   
2  Government expenditure on education, total (% ...  SE.XPD.TOTL.GD.ZS   NaN   
3  Government expenditure on education, total (% ...  SE.XPD.TOTL.GD.ZS   NaN   
4  Government expenditure on education, total (% ...  SE.XPD.TOTL.GD.ZS   NaN   

   1961  1962  1963  1964  1965  ...      2016      2017      2018      2019  \
0   NaN   NaN   NaN   NaN   NaN  ...  5.491360  4.455820  4.548764  4.435037   
1   NaN   NaN   NaN   NaN   NaN  ...  4.692000  4.430510  4.739750  4.511410   
2   NaN   NaN   NaN   NaN   NaN  ...  4.543970  4.343190       NaN       NaN   
3   NaN   NaN   NaN   NaN   NaN  ...  2.615035  3.296630  3.051252  3.047399   
4   NaN   NaN   NaN   NaN   NaN  ...  2.754937  2.466879  2.183513  2.073064   

       2020      2021      2022      2023  2024  Unnamed: 69  
0       NaN  3.618558       NaN       NaN   NaN          NaN  
1  4.090565  4.368379  3.697668  3.962293   NaN          NaN  
2       NaN       NaN       NaN       NaN   NaN          NaN  
3  3.398741  3.096926  2.891687  3.215620   NaN          NaN  
4  2.667447  2.297197  2.385359  2.512737   NaN          NaN  

[5 rows x 70 columns]

In [4]:
lifeDf.head()

Country Name Country Code  \
0                        Aruba          ABW   
1  Africa Eastern and Southern          AFE   
2                  Afghanistan          AFG   
3   Africa Western and Central          AFW   
4                       Angola          AGO   

                            Indicator Name  Indicator Code       1960  \
0  Life expectancy at birth, total (years)  SP.DYN.LE00.IN  64.049000   
1  Life expectancy at birth, total (years)  SP.DYN.LE00.IN  44.169658   
2  Life expectancy at birth, total (years)  SP.DYN.LE00.IN  32.799000   
3  Life expectancy at birth, total (years)  SP.DYN.LE00.IN  37.779636   
4  Life expectancy at birth, total (years)  SP.DYN.LE00.IN  37.933000   

        1961       1962       1963       1964       1965  ...       2016  \
0  64.215000  64.602000  64.944000  65.303000  65.615000  ...  75.540000   
1  44.468838  44.877890  45.160583  45.535695  45.770723  ...  62.167981   
2  33.291000  33.757000  34.201000  34.673000  35.124000  ...  62.646000   
3  38.058956  38.681792  38.936918  39.194580  39.479784  ...  56.392452   
4  36.902000  37.168000  37.419000  37.704000  37.968000  ...  61.619000   

        2017       2018       2019       2020       2021       2022  \
0  75.620000  75.880000  76.019000  75.406000  73.655000  76.226000   
1  62.591275  63.330691  63.857261  63.766484  62.979999  64.487020   
2  62.406000  62.443000  62.941000  61.454000  60.417000  65.617000   
3  56.626439  57.036976  57.149847  57.364425  57.362572  57.987813   
4  62.122000  62.622000  63.051000  63.116000  62.958000  64.246000   

        2023  2024  Unnamed: 69  
0  76.353000   NaN          NaN  
1  65.146154   NaN          NaN  
2  66.035000   NaN          NaN  
3  58.855722   NaN          NaN  
4  64.617000   NaN          NaN  

[5 rows x 70 columns]

In [5]:
healthDf.head()

Entity Code  Year  Public health expenditure as a share of GDP
0  Argentina  ARG  1880                                          0.0
1  Argentina  ARG  1890                                          0.0
2  Argentina  ARG  1900                                          0.0
3  Argentina  ARG  1910                                          0.0
4  Argentina  ARG  1920                                          0.0

In [6]:
eduDf_2019 = eduDf[["Country Name", "2019"]].copy()
eduDf_2019 = eduDf_2019.rename(columns={
    "Country Name": "Country",
    "2019": "EducationSpending"
})
eduDf_2019.head()

Country  EducationSpending
0                        Aruba           4.435037
1  Africa Eastern and Southern           4.511410
2                  Afghanistan                NaN
3   Africa Western and Central           3.047399
4                       Angola           2.073064

In [7]:
lifeDf_2019 = lifeDf[["Country Name", "2019"]].copy()
lifeDf_2019 = lifeDf_2019.rename(columns={
    "Country Name": "Country",
    "2019": "LifeExpectancy"
})
lifeDf_2019.head()

Country  LifeExpectancy
0                        Aruba       76.019000
1  Africa Eastern and Southern       63.857261
2                  Afghanistan       62.941000
3   Africa Western and Central       57.149847
4                       Angola       63.051000

In [8]:
healthDf_2019 = healthDf[healthDf["Year"] == 2019][
    ["Entity", "Public health expenditure as a share of GDP"]
].copy()

healthDf_2019 = healthDf_2019.rename(columns={
    "Entity": "Country",
    "Public health expenditure as a share of GDP": "HealthSpending"
})
healthDf_2019.head()

Country  HealthSpending
25   Argentina           6.245
95   Australia           7.349
166    Austria           7.906
205    Belgium           8.076
236     Brazil           3.930

In [9]:
df = eduDf_2019.merge(lifeDf_2019, on="Country") \
               .merge(healthDf_2019, on="Country")
df = df.dropna()

In [10]:
df.head()

Country  EducationSpending  LifeExpectancy  HealthSpending
0  Argentina            4.77165       76.847000           6.245
2    Austria            5.23678       81.895122           7.906
3    Belgium            6.32381       81.995122           8.076
4   Bulgaria            4.21429       75.112195           4.313
5     Brazil            5.96347       75.809000           3.930

#### regression

In [11]:
X = df[["EducationSpending", "HealthSpending"]]
y = df["LifeExpectancy"]

In [12]:
from sklearn.linear_model import LinearRegression

In [13]:
model = LinearRegression()
model.fit(X, y)

LinearRegression()

In [14]:
coef_df = pd.DataFrame({
    "Variable": X.columns,
    "Coefficient": model.coef_
})

intercept = model.intercept_

coef_df, intercept

(            Variable  Coefficient
 0  EducationSpending     0.140736
 1     HealthSpending     1.006350,
 np.float64(72.75616047331856))

LifeExpectancyi​ = β0​ + β1​ х EducationSpendingi ​+ β2​ х HealthSpendingi ​+ εi

LifeExpectancy ​= 72.8 ​+ 0.14 х EducationSpending + 1.006 х HealthSpending

intercept represents the theoretical life expectancy when all explanatory variables are equal to zero and is equal to 72-73 years.

In [15]:
r2 = model.score(X, y)
r2

0.37832216777273364

The coefficient of determination R² equals 0.38, which means that approximately 38% of the variation in life expectancy across countries is explained by education and healthcare expenditures included in the model.

#### correlation

Multicollinearity refers to a situation where independent variables in a regression model are highly correlated with each other, which can make coefficient estimates unstable and difficult to interpret.

In [ ]:
from statsmodels.stats.outliers_influence import variance_inflation_factor

X = df[["EducationSpending", "HealthSpending"]].dropna()

vif = pd.DataFrame()
vif["Variable"] = X.columns
vif["VIF"] = [variance_inflation_factor(X.values, i) for i in range(X.shape[1])]
vif

Variable      VIF
0  EducationSpending  8.98379
1     HealthSpending  8.98379

In [18]:
X = df[["EducationSpending", "HealthSpending"]]

X.corr()

EducationSpending  HealthSpending
EducationSpending           1.000000        0.452715
HealthSpending              0.452715        1.000000